pkg/pgp/pgp.go

```diff
@@ -138,9 +103,40 @@ func DecodePgpKey(pgpKey, path string) (*openpgp.Entity, error) {
 
 // DecodeAndArmorBase64Entity decodes a base64 encoded entity and armors it.
 func DecodeAndArmorBase64Entity(encodedEntity string, armorType string) (string, error) {
-	decodedEntity, err := base64.StdEncoding.DecodeString(encodedEntity)
+	decodedEntity, err := decodeAndValidatePGPkey(encodedEntity)
 	if err != nil {
 		return "", err
 	}
 	return parmor.ArmorWithType([]byte(decodedEntity), armorType)
 }
+
+// decodeAndValidatePGPkey PGP key is a wrapper function
+// around golang standard base64 package
+// which will validate keys with checksum as well as non checksum and return decoded bytes from key part
+func decodeAndValidatePGPkey(encodedKeyData string) ([]byte, error) {
+	// check if the string is standard PGP key with base64 encoding WITHOUT checksum
+	decodedKeyData, err := base64.StdEncoding.DecodeString(encodedKeyData)
+	if err != nil {
+		// check if keys is encoded using OpenPGP's Radix-64 encoding
+		// save the actual error
+		decodeErr := err
+		// create ascii armored gpg key string by adding
+		// ------ BEGIN -------
+		// encodedKeyData (base64key + '=' + checksum)
+		// ------  END  -------
+
+		// remove trailing newLines if any
+		encodedKeyData = strings.Trim(encodedKeyData, "\n")
+		pgpKey := fmt.Sprintf("%s\n\n%s\n%s", keyArmor("BEGIN"), encodedKeyData, keyArmor("END"))
```

Comment:
Does the account-notifier support this?

Reply:
@janboll  i have added a testcase for key data with checksum...Also when gql get query pulls in key data in a format having each line with 64 charcters seperated with `\n` as a single string which supports internal decode functions

Reply:
thank you, looking good